app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SalesController < ApplicationController
+  def index
+    @sales = Vendor.find(params[:vendor_id]).sales
```

Comment:
Same deal here with `before_action` as in `MarketsController` and `ProductsController`.